shared/modules/telemetry_module.go

```diff
@@ -0,0 +1,53 @@
+package modules
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type TelemetryModule interface {
+	Module
+
+	GetTimeSeriesAgent() TimeSeriesAgent
+	GetEventMetricsAgent() EventMetricsAgent
+}
+
+// Interface for time series agent (prometheus)
+type TimeSeriesAgent interface {
+	// Counters
+
+	// Registers a counter by name
+	CounterRegister(name string, description string)

```

Comment:
No. At the moment, if you try to re-register anything you've already registered, we ignore the fact and carry on cuz the job was done.

I see what you are saying, but this can be looked at as an improvement. When we are happy with the base functionality, we can revise code behavior however we see fit.